testctrl/svc/orch/executor_test.go

```diff
@@ -0,0 +1,213 @@
+
+	components := []*types.Component{server, client, driver}
+	session := types.NewSession(driver, components[:2], nil)
+
+	e := newKubeExecutor(0, fakePodInf, nil)
+	eventChan := make(chan *PodWatchEvent)
+	e.eventChan = eventChan
+	e.session = session
+
+	go func() {
+		for _, c := range components {
+			eventChan <- &PodWatchEvent{
+				SessionName:   session.Name,
+				ComponentName: c.Name,
+				Pod:           nil,
+				PodIP:         "127.0.0.1",
+				Health:        Ready,
+				Error:         nil,
+			}
+		}
+	}()
+
+	if err := e.provision(); err != nil {
+		t.Fatalf("unexpected error in provision: %v", err)
+	}
+
+	pods, err := listPods(t, fakePodInf)
+	if err != nil {
+		t.Fatalf("could not list pods from provision: %v", err)
+	}
+
+	expectedNames := []string{driver.Name, server.Name, client.Name}
+	for _, en := range expectedNames {
+		found := false
+
+		for _, pod := range pods {
+			if strings.Compare(pod.ObjectMeta.Name, en) == 0 {
+				found = true
+			}
+		}
+
+		if !found {
+			t.Errorf("provision did not create pod for component %v", en)
+		}
+	}
+
+	// test provision with an unsuccessful pod
+	fakePodInf = newFakePodInterface(t)
+
+	driver = types.NewComponent(testContainerImage, types.DriverComponent)
+	server = types.NewComponent(testContainerImage, types.ServerComponent)
+	client = types.NewComponent(testContainerImage, types.ClientComponent)
+
+	components = []*types.Component{server, client, driver}
+	session = types.NewSession(driver, components[:2], nil)
+
+	e = newKubeExecutor(0, fakePodInf, nil)
+	eventChan = make(chan *PodWatchEvent)
+	e.eventChan = eventChan
+	e.session = session
+
+	go func() {
+		for _, c := range components[:2] {
+			eventChan <- &PodWatchEvent{
+				SessionName:   session.Name,
+				ComponentName: c.Name,
+				Pod:           nil,
+				PodIP:         "127.0.0.1",
+				Health:        Ready,
+				Error:         nil,
+			}
+		}
+
+		eventChan <- &PodWatchEvent{
+			SessionName:   session.Name,
+			ComponentName: components[2].Name,
+			Pod:           nil,
+			PodIP:         "",
+			Health:        Failed,
+			Error:         nil,
+		}
+	}()
+
+	if err := e.provision(); err == nil {
+		t.Errorf("expected error for failing pod, but returned nil")
+	}
+}
+
+func TestKubeExecutorMonitor(t *testing.T) {
+	cases := []struct {
+		description string
+		event       *PodWatchEvent
+		errors      bool
+	}{
+		{
+			description: "success event received",
+			event:       &PodWatchEvent{Health: Succeeded},
+			errors:      false,
+		},
+		{
+			description: "failure event received",
+			event:       &PodWatchEvent{Health: Failed},
+			errors:      true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.description, func(t *testing.T) {
+			fakePodInf := newFakePodInterface(t)
+
+			driver := types.NewComponent(testContainerImage, types.DriverComponent)
+			server := types.NewComponent(testContainerImage, types.ServerComponent)
+			client := types.NewComponent(testContainerImage, types.ClientComponent)
```

Comment:
I think the order should match the components slice. I'll reorder:

```suggestion
			server := types.NewComponent(testContainerImage, types.ServerComponent)
			client := types.NewComponent(testContainerImage, types.ClientComponent)
			driver := types.NewComponent(testContainerImage, types.DriverComponent)
```

Reply:
Will fix later when tests are moved to tables